erizo_controller/erizoClient/src/webrtc-stacks/PeerConnectionFsm.js

```diff
@@ -0,0 +1,95 @@
+/* global */
+
+import StateMachine from '../../lib/state-machine';
+import Logger from '../utils/Logger';
+
+const activeStates = ['initial', 'failed', 'stable'];
+const HISTORY_SIZE_LIMIT = 200;
+// FSM
+const PeerConnectionFsm = StateMachine.factory({
+  init: 'initial',
+  transitions: [
+    { name: 'create-offer', from: activeStates, to: 'stable' },
+    { name: 'add-ice-candidate', from: activeStates, to: function nextState() { return this.state; } },
+    { name: 'process-answer', from: activeStates, to: 'stable' },
+    { name: 'process-offer', from: activeStates, to: 'stable' },
+    { name: 'negotiate-max-bw', from: activeStates, to: 'stable' },
+    { name: 'add-stream', from: activeStates, to: function nextState() { return this.state; } },
+    { name: 'remove-stream', from: activeStates, to: function nextState() { return this.state; } },
+    { name: 'close', from: activeStates, to: 'closed' },
+    { name: 'error', from: '*', to: 'failed' },
+  ],
+  data: function data(baseStackCalls) {
+    return { baseStackCalls,
+      history: [],
+    };
+  },
+  methods: {
+    getHistory: function getHistory() {
+      return this.history;
+    },
+    onBeforeClose: function onBeforeClose(lifecycle) {
```

Comment:
I'd leave a blank line between functions here too. 
Also, is it possible to have lambdas here?

```js
methods: {
  getHistory: () => {
  },

  onBeforeClose: () => {
  },
...
};
``` 
Possibly not due to the use of `this`, but I'm not sure.

Reply:
Yes, it's because the use of this in the state-machine functions. Agree on the use of blank lines 👍